### ISSUE TITLE
refactor-equipment.php - "Laufschuhe"

### ISSUE THREAD

**Issue by Wenerus** [Runalyze/Runalyze#1531]
When i call the script "refactor-equipment.php" again after a successful runy they add every time a new category (e.g. "Laufschuhe"). Now i have 1x "Ausrüstung" and 3x "Laufschuhe"

------

Wenn ich das Script "refactor-equipment.php" erneut aufrufe, erstellt mir dieses immer wieder eine neue Kategorie "Laufschuhe". Somit habe ich nun 1x "Ausrüstung" und 3x "Laufschuhe". An sich kein Problem, kann dies aber abgefangen werden?

**Comment by laufhannes** [comment]
Have there been any errors while executing the script?

The script does only start as long as the old `runalyze_shoe` table is present. Once the script has finished, the table is dropped so there is no chance to run the script a second time after it has completed.

For large databases with a lot of accounts, the script is able to run multiple times (by definining `LIMIT`). But even there, the script adds a column `refactored` at the first start and only accounts not refactored so far are refactored.

**Comment by Wenerus** [comment]
The following message appears when i start the script:
------
Table runalyze_account has column 'refactored'.
- refactored: 0/1
------
I have only one account and a small DB.

**Comment by mipapo** [comment]
@Wenerus  Does appear a "." or  something like this after" - refactored: 0/1":
. done;<br>
````<br>
Time: 0.032745838165283s<br>
Memory peak: 339544B<br>
<br>
... <a href="javascript:location.reload()">reload to continue</a>You are done. All rows are refactored.<br>
All unused tables (shoe, clothes) have been dropped.<br>
All unused columns from training (shoeid, clothes) have been dropped.<br>
<br>
Remember to unset your credentials within this file.<br>
(Or simply delete this file if you are not working on our git repository)<br>

````

**Comment by Wenerus** [comment]
@mipapo nope - only 

---

Table runalyze_account has column 'refactored'.
- refactored: 0/1

---

noting more or less - i will download the nightly and retry it from the scratch

**Comment by laufhannes** [comment]
Please take a look at your server logs or enable error reporting. There must be some error that stops the script.

**Comment by Wenerus** [comment]
First run after the update script
>Table runalyze_account has column 'refactored'.
>- refactored: 0/1
>
>. done;
>
>Time: 0.2011342048645s
>Memory peak: 416168B
>
>... reload to continueYou are done. All rows are refactored.
>All unused tables (shoe, clothes) have been dropped.
>All unused columns from training (shoeid, clothes) have been dropped.
>
>Remember to unset your credentials within this file.
>(Or simply delete this file if you are not working on our git repository)

after the reload
>Added column 'refactored' to table account.
>Table runalyze_account has column 'refactored'.
>- refactored: 0/1

@laufhannes  @mipapo  - netcup write their logs only ones a day

**Comment by Wenerus** [comment]
>[Mon Oct 19 16:55:29 2015] [warn] [client XX.XX.XX.XX] mod_fcgid: stderr: PHP Fatal error:  Uncaught exception 'PDOException' with message 'SQLSTATE[42S02]: Base table or view not found: 1146 Table 'DB.runalyze_shoe' doesn't exist' in /var/www/vhosts/SERVER/runalyzetest.runzen.de/refactor-equipment.php:141, referer: https://runalyzetest.runzen.de/refactor-equipment.php

>[Mon Oct 19 16:55:29 2015] [warn] [client XX.XX.XX.XX] mod_fcgid: stderr: Stack trace:, referer: https://runalyzetest.runzen.de/refactor-equipment.php

>[Mon Oct 19 16:55:29 2015] [warn] [client XX.XX.XX.XX] mod_fcgid: stderr: #0 /var/www/vhosts/SERVER/runalyzetest.runzen.de/refactor-equipment.php(141): PDO->query('SELECT `id`, `n...'), referer: https://runalyzetest.runzen.de/refactor-equipment.php

>[Mon Oct 19 16:55:29 2015] [warn] [client XX.XX.XX.XX] mod_fcgid: stderr: #1 {main}, referer: https://runalyzetest.runzen.de/refactor-equipment.php

>[Mon Oct 19 16:55:29 2015] [warn] [client XX.XX.XX.XX] mod_fcgid: stderr:   thrown in /var/www/vhosts/SERVER/runalyzetest.runzen.de/refactor-equipment.php on line 141, referer: https://runalyzetest.runzen.de/refactor-equipment.php

**Comment by laufhannes** [comment]
As it is stated after first run, `You are done. All rows are refactored.` - No need to run it again.

But the second run should stop in https://github.com/Runalyze/Runalyze/blob/master/refactor-equipment.php#L78-L83.

**Comment by laufhannes** [comment]
Ok, I guess that check was wrong. Should be fixed with 6326fecdc0b8e7f15860b0776dfdd33fe2b76818.
